arrays_test.go

```diff
@@ -70,30 +70,29 @@ const bands = `{
 }`
 
 func TestArrayCorrectness(t *testing.T) {
-	// only pattern3 should match
-	pattern1 := `{"bands": { "members": { "given": [ "Mick" ], "surname": [ "Strummer" ] } } }`
-	pattern2 := `{"bands": { "members": { "given": [ "Wata" ], "role": [ "drums" ] } } }`
-	pattern3 := `{"bands": { "members": { "given": [ "Wata" ], "role": [ "guitar" ] } } }`
+	// only wataGuiterPattern should match
```

Comment:
a bit curious why the renaming. More readable for sure!

Reply:
I had a bug which failed this test and it was hard for me to debug it because of the names `pattern1` / `pattern2` etc so I did those changes.

BTW, I think Quamina testing will benefit by adding good assertions library (like https://github.com/stretchr/testify)

Reply:
I'm used to that kind of thing from my years in Java-land. Hmm, I don't find writing tests of the form
```go
if x != nil {
  t.Error("whatever")
}
```

But I guess the advantage is that with the assertions and so on, the tests are more readable? That seems like a good argument.

Reply:
The benefit of assertions library is increasing readability of tests, in addition to better error messages when tests fail (for example: comparing two arrays, it will output both arrays to the screen with delta).

I wrote this as a note to myself, to check an assertion library for Quamina and purpose it, I'll make some examples so we can have an elaborate discussion on a draft PR.